base/abstractarray.jl

```diff
@@ -177,12 +177,15 @@ function _strides{M,T,N}(out::NTuple{M}, A::AbstractArray{T,N})
 end
 
 function isassigned(a::AbstractArray, i::Int...)
-    # TODO
     try
         a[i...]
         true
-    catch
-        false
+    catch e
+        if isa(e, BoundsError) || isa(e, UndefRefError)
```

Comment:
But yeah, why not just use `checkbounds` for the bounds check part? I

There doesn't seem to be anyway of checking if the field is undef without just trying to access it since that error is thrown from the runtime?